### ISSUE TITLE
not supported [10.0.14393.3986]

### ISSUE THREAD

**Issue by enigmen** [stascorp/rdpwrap#1183]
![изображение](https://user-images.githubusercontent.com/39157390/95959675-da072180-0e13-11eb-999c-301f121a7084.png)
autoupdate not helping(

**Comment by samthelion17** [comment]
Get the updated rdpwrap.ini file from [here](https://github.com/sebaxakerhtc/rdpwrap.ini/blob/master/rdpwrap.ini) and place it in the same folder as RDPWinst. Next uninstall your current version and reinstall with the new rdpwrap.ini.

**Comment by hbakiewicz** [comment]
I've tested the version of this rdpwrap.ini and for this particular version it doesn't work 
also I do not see 10.0.14393.3986 in [this ini file ](https://github.com/sebaxakerhtc/rdpwrap.ini/blob/master/rdpwrap.ini )

**Comment by samthelion17** [comment]
You're right, there's no [10.0.14393.3986] section in that file, sorry. I've done a little research and I think you have tree options:
1. Find out how to to patch the file yourself
2. Ask/Wait for someone to make a patch
3. Update Windows to a version supported by any of these rdpwrap.ini files

**Comment by jjwhite77** [comment]
samthelion,
is there any chance you are going to add the updated section to the ini file ?

**Comment by samthelion17** [comment]
Sorry buddy but I'm just a user, i have absolutely no idea on how to do so.

**Comment by jjwhite77** [comment]
Ok, Thanks
Do you know where I should be looking for the INI update ?

**Comment by samthelion17** [comment]
Well, [this](https://github.com/sebaxakerhtc/rdpwrap.ini) repo was modified 2 days ago, and the modified file was the rdpwrap.ini, so i assume it is one of the most updated. Maybe you can ask the owner to patch the file for your version of Windows.

**Comment by CrisRmz** [comment]
So any have an idea of how to fix it? :(

**Comment by Adeshwar03** [comment]
Same problem

**Comment by Razorvu** [comment]
waiting for the solution

**Comment by enigmen** [comment]
workaround - uninstall update KB4580346

**Comment by hbakiewicz** [comment]
I used this solution : [link ](http://woshub.com/how-to-allow-multiple-rdp-sessions-in-windows-10/)
check the last comment,  where "KUBAT" provided solution how to patch the dll.
for me it worked

**Comment by sweettweaks** [comment]
Hello same code as the previous one [10.0.14393.3503]
just copy paste and change the code to [10.0.14393.3986]
for the lazy ones here the code to paste at the end on the .ini file (and dont forget to leave a blanc line after the code or it will not work)


[10.0.14393.3986]
LocalOnlyPatch.x86=1
LocalOnlyOffset.x86=A6578
LocalOnlyCode.x86=jmpshort
LocalOnlyPatch.x64=1
LocalOnlyOffset.x64=8D8A1
LocalOnlyCode.x64=jmpshort
SingleUserPatch.x86=1
SingleUserOffset.x86=36CE5
SingleUserCode.x86=nop
SingleUserPatch.x64=1
SingleUserOffset.x64=1B6A4
SingleUserCode.x64=Zero
DefPolicyPatch.x86=1
DefPolicyOffset.x86=31209
DefPolicyCode.x86=CDefPolicy_Query_eax_ecx
DefPolicyPatch.x64=1
DefPolicyOffset.x64=F185
DefPolicyCode.x64=CDefPolicy_Query_eax_rcx
SLInitHook.x86=1
SLInitOffset.x86=45912
SLInitFunc.x86=New_CSLQuery_Initialize
SLInitHook.x64=1
SLInitOffset.x64=22C80
SLInitFunc.x64=New_CSLQuery_Initialize

[10.0.14393.3986-SLInit]
bInitialized.x86      =C2F94
bServerSku.x86        =C2F98
lMaxUserSessions.x86  =C2F9C
bAppServerAllowed.x86 =C2FA0
bRemoteConnAllowed.x86=C2FA4
bMultimonAllowed.x86  =C2FA8
ulMaxDebugSessions.x86=C2FAC
bFUSEnabled.x86       =C2FB0

bServerSku.x64        =E73D0
lMaxUserSessions.x64  =E73D4
bAppServerAllowed.x64 =E73D8
bInitialized.x64      =E8470
bRemoteConnAllowed.x64=E8474
bMultimonAllowed.x64  =E8478
ulMaxDebugSessions.x64=E847C
bFUSEnabled.x64       =E8480

**Comment by enigmen** [comment]
a try change 3503 to 3986, restart rdp but whatever not work

**Comment by Razorvu** [comment]
> Hello same code as the previous one [10.0.14393.3503]
> just copy paste and change the code to [10.0.14393.3986]
> for the lazy ones here the code to paste at the end on the .ini file (and dont forget to leave a blanc line after the code or it will not work)
> 
> [10.0.14393.3986]
> LocalOnlyPatch.x86=1
> LocalOnlyOffset.x86=A6578
> LocalOnlyCode.x86=jmpshort
> LocalOnlyPatch.x64=1
> LocalOnlyOffset.x64=8D8A1
> LocalOnlyCode.x64=jmpshort
> SingleUserPatch.x86=1
> SingleUserOffset.x86=36CE5
> SingleUserCode.x86=nop
> SingleUserPatch.x64=1
> SingleUserOffset.x64=1B6A4
> SingleUserCode.x64=Zero
> DefPolicyPatch.x86=1
> DefPolicyOffset.x86=31209
> DefPolicyCode.x86=CDefPolicy_Query_eax_ecx
> DefPolicyPatch.x64=1
> DefPolicyOffset.x64=F185
> DefPolicyCode.x64=CDefPolicy_Query_eax_rcx
> SLInitHook.x86=1
> SLInitOffset.x86=45912
> SLInitFunc.x86=New_CSLQuery_Initialize
> SLInitHook.x64=1
> SLInitOffset.x64=22C80
> SLInitFunc.x64=New_CSLQuery_Initialize
> 
> [10.0.14393.3986-SLInit]
> bInitialized.x86 =C2F94
> bServerSku.x86 =C2F98
> lMaxUserSessions.x86 =C2F9C
> bAppServerAllowed.x86 =C2FA0
> bRemoteConnAllowed.x86=C2FA4
> bMultimonAllowed.x86 =C2FA8
> ulMaxDebugSessions.x86=C2FAC
> bFUSEnabled.x86 =C2FB0
> 
> bServerSku.x64 =E73D0
> lMaxUserSessions.x64 =E73D4
> bAppServerAllowed.x64 =E73D8
> bInitialized.x64 =E8470
> bRemoteConnAllowed.x64=E8474
> bMultimonAllowed.x64 =E8478
> ulMaxDebugSessions.x64=E847C
> bFUSEnabled.x64 =E8480

thanks) work

**Comment by Adeshwar03** [comment]
I can confirm that it is working. Either add the lines above or Use Addinityv full code to copy paste in RDPwrap.ini and you will have it working.

**Comment by enigmen** [comment]
hi, i run autoupdate, and see RDPConf
![изображение](https://user-images.githubusercontent.com/39157390/96221526-cd153a00-0f9b-11eb-9ee4-59b0b42962fd.png)
but when trying to connect rdp - error
![изображение](https://user-images.githubusercontent.com/39157390/96221661-fdf56f00-0f9b-11eb-94f3-20acdcc3f26d.png)
and after that if you look RDPConf, then
![изображение](https://user-images.githubusercontent.com/39157390/96221766-29785980-0f9c-11eb-9657-7f7b0119811f.png)
I removed the rdpwrap, installed it again, rebooting pc, all the same the same. I don't understand why it doesn't work.

**Comment by affinityv** [comment]
@enigmen Windows Defender or some other "av security" "protection" product may be interfering, add exception.  Make sure the new INI file is in place, then do `RDPWInst.exe -r` as admin user.

**Comment by sweettweaks** [comment]
hello
have left a blank (empty) line at the end of the .ini file? without that the service stops.
to edit the .ini file just run a text edit as admin (notepad for example) than once opened as admin, go on file, open navigate to the ini file ad the code, add an empty line and register

restart termserv (on cmd prompt as **admin** type **net stop termserv** to stop the service, than **net start termserv** to start the service)

this should work

keep checking running and listening state with rdpconf

**Comment by enigmen** [comment]
> 
> 
> @enigmen Windows Defender or some other "av security" "protection" product may be interfering, add exception. Make sure the new INI file is in place, then do `RDPConf.exe` as admin user.

windows defender and security protection is off, yes run as admin user.

**Comment by enigmen** [comment]
> 
> 
> hello
> have left a blank (empty) line at the end of the .ini file? without that the service stops.
> to edit the .ini file just run a text edit as admin (notepad for example) than once opened as admin, go on file, open navigate to the ini file ad the code, add an empty line and register
> 
> restart termserv (on cmd prompt as **admin** type **net stop termserv** to stop the service, than **net start termserv** to start the service)
> 
> this should work
> 
> keep checking running and listening state with rdpconf

after autoupdate and net stop - net start
![изображение](https://user-images.githubusercontent.com/39157390/96236477-26399980-0fad-11eb-94ea-d31ed2d9bb15.png)

**Comment by enigmen** [comment]
i just replaced termsrv.dll ver 10.0.14393.3986 to 10.0.14393.3503 and work)

**Comment by sweettweaks** [comment]
yep, messing with the files is not the best way to go, i just keep the original ones

**Comment by McSon1** [comment]
https://github.com/stascorp/rdpwrap/issues/1183#issuecomment-709918807
а подскажите где взять termsrv.dll ver 10.0.14393.3503 
чтобы сменить 10.0.14393.3986

**Comment by antiRad-web** [comment]
Sorry to confirm that after the ini-patch to 14393.3986 provided by @affinityv - although the RDPWrapper configuration looks like fully supported - the remote desktop service terminates if a user tries to log in. Same result after restarting this service as soon as users try to log in via RDP again. I turned off Antivirus products, Defender, etc. before.
Before uninstalling the whole October Update, I rolled back to 14393.3503 version of termsrv.dll only, but concurrent RDP sessions are then not possible anymore. Users are forced to choose a running session to terminate before they are allowed to log in.
I now finally ended up in uninstalling the October 2020 Update.
I assume that the ini-patch details from 14393.3503 cannot completely transferred to the ini-patch for 14393.3986 without any changes.

**Comment by Adeshwar03** [comment]
I agree with AntiRAD it worked once or twice.. later it is broken.

**Comment by enigmen** [comment]
![изображение](https://user-images.githubusercontent.com/39157390/96407398-5a51cc00-11f2-11eb-8bc8-b94ef93ead0d.png)
[termsrv.zip](https://github.com/stascorp/rdpwrap/files/5399423/termsrv.zip)
just replaced  dll, and no problem:)

**Comment by Adeshwar03** [comment]
This results in a single session.. and disconnects the other user.

**Comment by sweettweaks** [comment]
well i have several machines working just fine on win 10 and win server 2016

i would suggest: 

revert to the original dll files (and please don't mess with the dll's)
make sure that you save as utf-8 when you modify the .ini file
make sure you have an EMPTY LINE at the end of the .ini file.
check the dll file rights and, to be sure you have no other problems, run: **SFC /scannow** from an elevated cmd

if this does not work start praying :)

**Comment by Jetteroh** [comment]
I updated MS Server 2016 at last weekend. I got this 10.0.14393.3986 termsvr.dll - I have RDWrap AUTOUPDATEr, but it loads an older .INI file - so not supporting.
For the working of my RDWrap I deleted the last update. After reboot RDWrap working again! As temporary solution, this can help you.

**Comment by pravyroxor** [comment]
Solution posted by sweettweaks worked for me.
![rdpwrap](https://user-images.githubusercontent.com/17481609/96541727-e28eaa80-12a0-11eb-8956-0bdd1dc628c9.JPG)
[rdpwrap.zip](https://github.com/stascorp/rdpwrap/files/5406345/rdpwrap.zip)

**Comment by affinityv** [comment]
UTF-8 ???? Plain ordinary text I think.
sfc /scannow iwll likely be an issue if you are not using the "current" termsrv.dll

**Comment by Jetteroh** [comment]
pravyroxor:
> Solution posted by sweettweaks worked for me.
> [rdpwrap.zip](https://github.com/stascorp/rdpwrap/files/5406345/rdpwrap.zip)

This rdpwrap.ini not working well for me on MS Server 2016:
![image](https://user-images.githubusercontent.com/8670412/96924594-26f19480-14b3-11eb-9aea-6a8910036c94.png)

**Comment by pravyroxor** [comment]
> This rdpwrap.ini not working well for me on MS Server 2016:
Actually it works, but you can see the error - Listener is not listening. This is not problem of rdp wrapper...
Your RDP connections are disabled or blocked by firewall.
REG ADD "HKLM\SYSTEM\CurrentControlSet\Control\Terminal Server" /v fDenyTSConnections /t REG_DWORD /d 0 /f

**Comment by Jetteroh** [comment]
@pravyroxor 
This is a working environment. After I deleted this update and the VPS restarted, I can login through RDP. RDWrap is working again. (I have autoupdater scheduled at startup.)
Though I will do all the procedure again at weekend. I will check is the Windows update changing this registry value.

**Comment by Jetteroh** [comment]
pravyroxor:
> Solution posted by sweettweaks worked for me.
> [rdpwrap.zip](https://github.com/stascorp/rdpwrap/files/5406345/rdpwrap.zip)
Now I can confirm that this rdwrap.ini is working on MS Server 2016:
![image](https://user-images.githubusercontent.com/8670412/97799704-eff84d00-1c2f-11eb-954b-1e06c32ca4eb.png)
I had put it in my autoupdate.bat file and now working well. It may happen before that the editing of rdwrap.ini was not as good as it should.
Thanks!

**Comment by yjmoming** [comment]
> ![изображение](https://user-images.githubusercontent.com/39157390/96407398-5a51cc00-11f2-11eb-8bc8-b94ef93ead0d.png)
> [termsrv.zip](https://github.com/stascorp/rdpwrap/files/5399423/termsrv.zip)
> just replaced dll, and no problem:)

Thanks,it works.

**Comment by seefalke** [comment]
sweettweaks solution worked for me! A lot of thanks!!
